TrustModel.md

```diff
@@ -0,0 +1,426 @@
+
+#### Distribute
+The Distribute stage includes all preparations before the guest executes its first instructions, 
+and it will run some general and TEE-specific operations.
+- configuring the cluster with TEE
+- making the boot image available
+- taking initial measurements
+- provision secrets.
+- allocation of guest resource
+- loading guest fw and configuration occurs 
+- TEE-specific steps relating to preparing attestation/secrets etc
+
+The involved components include the ones that directly associated with the
+protections that TEEs provide (e.g, tdx seam-loader, tdx module, psp/sev firmware and so on),
+Pre-OS (e.g, guest firmware, guest kernel, initrd and so on) and specific components (e.g,
+kata-agent, attestation-agent and so on). Also includes preparing secrets for delivery to the TEE.
+
+```The first instruction being executed within the trusted boot image happens in the next phase```
+
+#### Deploy
+  - pod sandbox preparation
+  - "pre flight checks" include attestation measurment checks
+  - secret provisioning linked to boot attestation
+  
+  ```Receiving the request to pull and deploy a container happens in the runtime phase```
+
+
+#### Runtime
+- post boot attestation and secret provisioning
+- procedure of image pulling, unpacking and 
+  launching.
+- the workload in the container starts to  run in TEEs.
+
+
+These execution phases are sufficient to describe the execution of any forms of confidential
+container. 
+
+---
+---
+
+## Goals and not-goals
+
+### Goals
+
+The commonality of goals between confidential container and confidential computing are to reduce
+the ability for the platform owner, operator and attacker to access data and code inside TEEs
+sufficiently such that this path is not an economically or logically viable attack during
+execution[^1]  5.1 Goal.
+
+Around the three execution stages of confidential container, the scope of the threat model of
+confidential container includes:
+
+<* ADD diagram*>
+
+- Pre-container stage
+  - Measure the TEE where confidential containers run, and store the measurements to a secure location.
+
+- Container launch stage
+  - Measure and attest the TEE where confidential containers run to prove the deployment and initialization of pre-container are valid and expected;
+  - Pull the encrypted and signed container image in TEE;
+  - Provision the runtime configurations through a secure and attested control plane with the protections of confidentiality and integrity.
+
+- Workload operation stage
+  - Provide the protections of confidentiality and integrity for the storage and transport of data derived from TEE outside the TEE instance;
+  - Keep the dynamic measurement of TEE where confidential containers run, and store the measurements to a secure location;
+  - Continuously launch the runtime attestations for TEE where confidential containers run according to the policy to ensure the TEEs are not comprised or out-of-date;
```

Comment:
Not available on all hardware platforms. I think one idea of Confidential Computing is to replace continuous measurement with initial measurement and strictly controlled boundaries between the trust domains. Not everyone thinks this is great, though.

Reply:
Because the only initial measurement is lack of flexibility against dynamic measurements. But it is still valuable to have a runtime attestation with a fixed initial measurement to allow the external services to authenticate the attester who initializes the runtime attestation, e.g, issue an attested heartbeat / healthy report.

Reply:
I agree that more measurements are better. Even SEV-SNP can't provide continuous dynamic measurement by default.

Reply:
I don't think "Continuously" makes sense here. Maybe "on-demand"? i.e. you can request an attestation from some platforms whenever you like (and some just at launch), but even then I think you are implying a polling mechanism.
nit: /comprised/compromised